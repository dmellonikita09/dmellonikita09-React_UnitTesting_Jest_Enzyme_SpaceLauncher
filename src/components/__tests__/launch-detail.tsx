import { mount } from 'enzyme';
import React from 'react';
import LaunchDetail from '../launch-detail';

describe('Launch Detail View', () => {
  it('renders without error', () => {
    const mockLaunch = {
      __typename: 'Launch',
      id: 1,

      rocket: {
        __typename: 'Rocket',
        id: 1,
        name: 'tester',
      }
    };    
    const wrapper=mount(<LaunchDetail />);
    expect(wrapper.find({mockLaunch}));
  });
    
  it('renders without error', () =>{
    const component= mount(
      <LaunchDetail
      id={'1'}
      site={'earth'}
      rocket={{ name: 'that one', type: 'big', __typename: 'Rocket', id: '1' }}
      />
    );
    expect(component).toMatchSnapshot();
  });  
});
