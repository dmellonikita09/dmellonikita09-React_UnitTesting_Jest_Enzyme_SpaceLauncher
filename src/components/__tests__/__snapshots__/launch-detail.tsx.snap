// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch Detail View renders without error 1`] = `
<LaunchDetail
  id="1"
  rocket={
    Object {
      "__typename": "Rocket",
      "id": "1",
      "name": "that one",
      "type": "big",
    }
  }
  site="earth"
>
  <Styled(div)
    style={
      Object {
        "backgroundImage": "url(iss.jpg)",
      }
    }
  >
    <div
      className="css-gfwuup"
      style={
        Object {
          "backgroundImage": "url(iss.jpg)",
        }
      }
    >
      <h3>
        that one
         (
        big
        )
      </h3>
      <h5>
        earth
      </h5>
    </div>
  </Styled(div)>
</LaunchDetail>
`;
